### PR TITLE
[cgal] update to 6.0

### PR DESCRIPTION
**Pull request by lrineau**
Update to [CGAL version 6.0](https://github.com/CGAL/cgal/releases/tag/v6.0). That new version has several breaking changes, but it seems nothing broke in vcpkg.

- [x] Changes comply with the [maintainer guide](https://github.com/microsoft/vcpkg-docs/blob/main/vcpkg/contributing/maintainer-guide.md).
- [x] SHA512s are updated for each updated download.
- [x] The "supports" clause reflects platforms that may be fixed by this new version.
- [x] Any fixed [CI baseline](https://github.com/microsoft/vcpkg/blob/master/scripts/ci.baseline.txt) entries are removed from that file.
- [x] Any patches that are no longer applied are deleted from the port's directory.
- [x] The version database is fixed by rerunning `./vcpkg x-add-version --all` and committing the result.
- [x] Only one version is added to each modified port's versions file.

### CODE REVIEW

File: ports/cgal/portfile.cmake [hunk 1]
```diff
@@ -4,16 +4,10 @@ vcpkg_buildpath_length_warning(37)
...
-    SHA512 943413bf3b94da066d47051b22f1c1b66a39a43dad865dd0e4912a9ae6508d9a490bcfe4ef7d662844e9a1ba6e552748fad785b52b2cce98133c2dfdfbfe1a4d
+    REF v6.0
+    SHA512 f61e608898d798b90ce07260928b682161f00e964b43b9876ef6604d10c30787a0814e13afde90f7d703efd6b83c61dd4a9d9f50d21068bd50c5c15f94b5755b
     HEAD_REF master
 )
-
-vcpkg_check_features(OUT_FEATURE_OPTIONS FEATURE_OPTIONS
-    FEATURES
-        qt WITH_CGAL_Qt5
-)
```
Comment (dg0yt) [inline ports/cgal/portfile.cmake:15]: This needs an update or replacement. There must be a strong coupling between the choice of features and Qt being enabled in the build.

If `WITH_CGAL_Qt5` is no longer set, it must also be removed from `MAYBE_UNUSED_VARIABLES` below.

File: ports/cgal/portfile.cmake [hunk 2]
```diff
@@ -4,16 +4,10 @@ vcpkg_buildpath_length_warning(37)
 vcpkg_from_github(
     OUT_SOURCE_PATH SOURCE_PATH
     REPO CGAL/cgal
-    REF v5.6.1
-    SHA512 943413bf3b94da066d47051b22f1c1b66a39a43dad865dd0e4912a9ae6508d9a490bcfe4ef7d662844e9a1ba6e552748fad785b52b2cce98133c2dfdfbfe1a4d
+    REF v6.0
```
Comment (dg0yt) [inline ports/cgal/portfile.cmake:7]:
```suggestion
    REF v${VERSION}
```

File: ports/cgal/vcpkg.json [hunk 3]
```diff
@@ -67,14 +66,9 @@
...
         "eigen3",
-        "qt5-3d",
-        {
-          "name": "qt5-base",
-          "default-features": false
-        },
-        "qt5-script",
-        "qt5-svg",
-        "qt5-xmlpatterns"
+        "qtbase",
```
Comment (dg0yt) [inline ports/cgal/vcpkg.json:69]:
```suggestion
        {
          "name": "qtbase",
          "default-features": false
        },
```
vcpkg will still install the default features by default, but the user may opt out (CLI, top-level manifest).

File: ports/cgal/PR-8523__CGAL_CGAL-protect_against_macro_free-GF.patch [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+```
+
+There is probably a macro named `free` somewhere.
+---
+ CGAL_Core/include/CGAL/CORE/ExprRep.h    | 4 ++--
+ CGAL_Core/include/CGAL/CORE/Impl.h       | 2 +-
+ CGAL_Core/include/CGAL/CORE/MemoryPool.h | 2 +-
+ CGAL_Core/include/CGAL/CORE/RealRep.h    | 2 +-
+ 4 files changed, 5 insertions(+), 5 deletions(-)
+
```
Comment (dg0yt) [inline ports/cgal/PR-8523__CGAL_CGAL-protect_against_macro_free-GF.patch:27]:
```suggestion
```
Please don't add this to this repo's history. The patch name has enough details.

**Review by dg0yt — COMMENTED** [review]
Given that the lib is header-only now:

- I wonder if the port still needs the `vcpkg_buildpath_length_warning`.
- I suggest to start the portfile with
  ~~~cmake
  set(VCPKG_BUILD_TYPE release)  # header-only
  ~~~

Please update the PR title to correctly reference `[cgal]`. This will appear in git history.

**Comment by lrineau** [comment]
@WangWeiLin-MV This pull-request is no longer a draft, and now passes the CI tests. 😌

**Pull request merged**